controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -210,15 +212,71 @@ func (r *OpenshiftAssistedControlPlaneReconciler) Reconcile(ctx context.Context,
 	result := ctrl.Result{}
 	if conditions.IsTrue(oacp, controlplanev1alpha2.KubeconfigAvailableCondition) {
 		// in case upgrade is still in progress, we want to requeue, however we also want to reconcile replicas
-		result, err = r.upgradeWorkloadCluster(ctx, cluster, oacp, log, pullsecret)
+		result, err = r.upgradeWorkloadCluster(ctx, cluster, oacp, architecture, pullsecret)
 		if err != nil {
 			return result, err
 		}
 	}
 	return result, r.reconcileReplicas(ctx, oacp, cluster)
 }
 
-func (r *OpenshiftAssistedControlPlaneReconciler) upgradeWorkloadCluster(ctx context.Context, cluster *clusterv1.Cluster, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, log logr.Logger, pullSecret []byte) (ctrl.Result, error) {
+func getArchitectureFromBootstrapConfigs(ctx context.Context, k8sClient client.Client, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) (string, error) {
+	defaultArch := "multi"
+
+	// if oacp is nil, return default arch
+	if oacp == nil {
+		return defaultArch, nil
+	}
+
+	// if no clusterName label available, return default arch
+	clusterName, ok := oacp.Labels[clusterv1.ClusterNameLabel]
+	if !ok {
+		return defaultArch, nil
+	}
+	labelSelector := map[string]string{
+		clusterv1.ClusterNameLabel: clusterName,
+	}
+	listOptions := []client.ListOption{
+		client.InNamespace(oacp.Namespace),
+		client.MatchingLabels(labelSelector),
+	}
+	var configList bootstrapv1alpha1.OpenshiftAssistedConfigList
+	if err := k8sClient.List(ctx, &configList, listOptions...); err != nil {
+		return "", err
+	}
+
+	architectures := make([]string, 0)
+	for _, config := range configList.Items {
+		architectures = append(architectures, config.Spec.CpuArchitecture)
```

Comment:
Sorry, I couldn't find in the codebase where this is set 😅 Where is `config.Spec.CpuArchitecture` set?

Reply:
It's never set, it has a default value https://github.com/openshift-assisted/cluster-api-agent/blob/master/bootstrap/api/v1alpha1/openshiftassistedconfig_types.go#L58

In any case, if it's empty, we consider them to be `multi` arch